frame/offences/src/lib.rs

```diff
@@ -187,14 +150,14 @@ where
 		let slash_perbill: Vec<_> = (0..concurrent_offenders.len())
 			.map(|_| new_fraction.clone()).collect();
 
-		let applied = Self::report_or_store_offence(
+		T::OnOffenceHandler::on_offence(
 			&concurrent_offenders,
 			&slash_perbill,
 			offence.session_index(),
 		);
 
 		// Deposit the event.
-		Self::deposit_event(Event::Offence(O::ID, time_slot.encode(), applied));
+		Self::deposit_event(Event::Offence(O::ID, time_slot.encode(), true));
```

Comment:
Once an Offence always succeed, should I remove the `applied` property from this event type?

```rust
pub enum Event {
	/// There is an offence reported of the given `kind` happened at the `session_index` and
	/// (kind-specific) time slot. This event is not deposited for duplicate slashes. last
	/// element indicates of the offence was applied (true) or queued (false)
	/// \[kind, timeslot, applied\].
	Offence(Kind, OpaqueTimeSlot, bool),
}
```

Reply:
yeah, good catch!